vpr/src/route/route_timing.cpp

```diff
@@ -342,7 +342,7 @@ bool try_timing_driven_route_tmpl(const t_router_opts& router_opts,
         device_ctx.rr_nodes,
         &device_ctx.rr_graph,
         device_ctx.rr_rc_data,
-        device_ctx.rr_switch_inf,
+        g_vpr_ctx.mutable_device().rr_graph_builder.rr_switch_inf(),
```

Comment:
Should not use a mutable object. A router function is not allowed to modify any routing resource graph.

Reply:
Also, we should consider to
- Return ``rr_switch_inf`` only through ``RRGraphView`` because ``RRGraphView`` is the object which client function should interface
- Avoid return raw data. Can you check how the functions in routing_timing.cpp use this rr_switch_inf? If they want to access any rr_switch, they should use ``RRSwitchId` and related APIs from ``RRGraphView``